src/wallet/sqlite.cpp

```diff
@@ -281,6 +282,16 @@ void SQLiteDatabase::Open()
             throw std::runtime_error(strprintf("SQLiteDatabase: Failed to set the wallet schema version: %s\n", sqlite3_errstr(ret)));
         }
     }
+
+    auto batch = MakeBatch();
+    if (!batch->Exists("uuid")) {
+        // UUID is missing (probably a new wallet)
+        std::vector<unsigned char> new_uuid_buf(32);
+        // We might not need strong random, but better safe than sorry
+        // Can't make weak random stronger later
+        GetStrongRandBytes(&new_uuid_buf[0], new_uuid_buf.size());
```

Comment:
This is actually not a rfc4122 UUID. It's just 256 random bits. Probably doesn't matter though.